src/parser/TextSearchQuery.cpp

```diff
@@ -0,0 +1,166 @@
+// Copyright 2025, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Felix Meisen (fesemeisen@outlook.de)
+
+#include "parser/TextSearchQuery.h"
+
+#include <absl/strings/str_split.h>
+
+#include "parser/MagicServiceIriConstants.h"
+
+namespace parsedQuery {
+
+// ____________________________________________________________________________
+void TextSearchQuery::addParameter(const SparqlTriple& triple) {
+  auto simpleTriple = triple.getSimple();
+  TripleComponent subject = simpleTriple.s_;
+  TripleComponent predicate = simpleTriple.p_;
+  TripleComponent object = simpleTriple.o_;
+
+  auto predString = extractParameterName(predicate, TEXT_SEARCH_IRI);
+
+  auto throwVariableException = [predString, subject, object]() {
+    if (!(subject.isVariable() && object.isVariable())) {
+      throw TextSearchException(
+          absl::StrCat("The predicate <", predString,
+                       "> needs a Variable as subject and one as object."));
+    }
+  };
+
+  if (predString == "text-search") {
+    if (!(subject.isVariable() && object.isVariable())) {
+      throw TextSearchException(
+          "The predicate <text-search> needs a Variable as subject and one as "
+          "object.");
+    }
+    if (configVarToConfigs_[object.getVariable()].textVar_.has_value()) {
+      throw TextSearchException(
+          "Each search should be linked to only on text Variable.");
+    }
+    configVarToConfigs_[object.getVariable()].textVar_ = subject.getVariable();
+  } else if (predString == "contains-word") {
+    if (!(subject.isVariable() && object.isLiteral())) {
+      throw TextSearchException(
+          "The predicate <contains-word> needs a Variable as subject and a "
+          "Literal as object.");
+    }
+    if (configVarToConfigs_[subject.getVariable()].isWordSearch_.has_value()) {
+      if (configVarToConfigs_[subject.getVariable()].isWordSearch_.value()) {
+        throw TextSearchException(
+            "One search should only contain one <contains-word>.");
+      } else {
+        throw TextSearchException(
+            "One search should contain either <contains-word> or "
+            "<contains-entity>.");
+      }
+    }
+    configVarToConfigs_[subject.getVariable()].isWordSearch_ = true;
+    std::string_view literal = object.getLiteral().toStringRepresentation();
+    literal = std::string(literal.substr(1, literal.size() - 2));
+    if (literal.empty()) {
+      throw TextSearchException(
+          "The predicate <contains-word> shouldn't have an empty literal as "
+          "object.");
+    }
+    configVarToConfigs_[subject.getVariable()].word_ = literal;
+  } else if (predString == "contains-entity") {
```

Comment:
You could split this up into several helper functions, one per predicate. This if-else is rather lengthy.